homeassistant/components/camera/rpicam.py

```diff
@@ -0,0 +1,334 @@
+            camera.record_video_on(duration)
+
+    def handle_video_off(service):
+        for camera in _get_cameras(service, all_if_no_id=True):
+            camera.record_video_off()
+
+    def handle_image(service):
+        for camera in _get_cameras(service, all_if_no_id=True):
+            camera.record_image()
+
+    def handle_exposure_mode(service):
+        mode = service.data.get(ATTR_MODE, None)
+        for camera in _get_cameras(service, all_if_no_id=True):
+            camera.change_exposure_mode(mode)
+
+    def handle_pantilt(service):
+        """Handle pan/tilt service call."""
+        view = service.data.get(ATTR_VIEW, None)
+        pan = service.data.get(ATTR_PAN, None)
+        tilt = service.data.get(ATTR_TILT, None)
+
+        for camera in _get_cameras(service, all_if_no_id=False):
+            camera.set_pantilt(view, pan, tilt)
+
+    hass.services.async_register(DOMAIN, SERVICE_REC_VIDEO_ON,
+                                 handle_video_on,
+                                 schema=SERVICE_REC_VIDEO_ON_SCHEMA)
+
+    hass.services.async_register(DOMAIN, SERVICE_REC_VIDEO_OFF,
+                                 handle_video_off,
+                                 schema=SERVICE_REC_VIDEO_OFF_SCHEMA)
+
+    hass.services.async_register(DOMAIN, SERVICE_REC_IMAGE,
+                                 handle_image,
+                                 schema=SERVICE_REC_IMAGE_SCHEMA)
+
+    hass.services.async_register(DOMAIN, SERVICE_EXPOSURE_MODE,
+                                 handle_exposure_mode,
+                                 schema=SERVICE_EXPOSURE_MODE_SCHEMA)
+
+    hass.services.async_register(DOMAIN, SERVICE_PANTILT,
+                                 handle_pantilt,
+                                 schema=SERVICE_PANTILT_SCHEMA)
+
+    async_add_entities([RpiCam(hass, config)])
+
+
+class RpiCam(GenericCamera):
+    """Raspberry PI camera support based rpi-cam-mqtt."""
+
+    def __init__(self, hass, device_info):
+        """Initialize an rpicam camera."""
+        super().__init__(hass, device_info)
+        self.cam = device_info.get(CONF_NAME)
+        _LOGGER.debug("Rpicam config found for \"%s\"", self.cam)
+        self._cmd_topic = "rpicam/{}".format(self.cam)
+        self._status_topic = "rpicam/{}/status".format(self.cam)
+        self._ptcmd_topic = "rpicam/{}/pt".format(self.cam)
+
+        self._qos = 1
+        self.rpi_status = None
+
+    @property
+    def state(self):
+        """Return the camera state."""
+        return self.rpi_status
+
+    @property
+    def is_on(self):
+        """Return true if on."""
+        return bool(self.rpi_status in ACTIVE_STATUSES)
+
+    @property
+    def is_recording(self):
+        """Return true if the device is recording."""
+        return bool(self.rpi_status in RECORDING_STATUSES)
+
+    def turn_off(self):
+        """Turn off camera."""
+        mqtt.publish(self.hass, self._cmd_topic, 'ru 0', self._qos)
```

Comment:
The device specific interface should be moved to a standalone library published on pypi.

The library can use the home assistant mqtt transport, or another mqtt transport, by accepting publish and subscribe callbacks.

See the mysensors component and its requirement pymysensors for an example of this.